addons/base/views.py

```diff
@@ -327,10 +327,14 @@ def get_auth(auth, **kwargs):
                 node_id=provider_settings.owner._id,
                 root_id=provider_settings.root_node._id,
             )
-    # If they haven't been set by version region, use the NodeSettings region
+    # If they haven't been set by version region, use the NodeSettings or Preprint directly
     if not (credentials and waterbutler_settings):
-        credentials = provider_settings.serialize_waterbutler_credentials()
-        waterbutler_settings = provider_settings.serialize_waterbutler_settings()
+        if provider_settings:
+            credentials = provider_settings.serialize_waterbutler_credentials()
+            waterbutler_settings = provider_settings.serialize_waterbutler_settings()
+        elif isinstance(node, Preprint):
```

Comment:
A bit smelly, but it looks like we're doing this type-checking elsewhere in the function, so perhaps this is something we rethink in the future.